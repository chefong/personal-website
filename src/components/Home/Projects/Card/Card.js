import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { FormattedMessage } from 'text-provider';
import PropTypes from 'prop-types';
import styles from './Card.module.scss';

export default function Card({
  nameId, descriptionId, githubLink,
}) {
  return (
    <div className="col-lg-4 col-md-6">
      <div className={styles.cardInfo}>
        <a href={githubLink} target="_blank" rel="noreferrer">
          <FontAwesomeIcon className={styles.github} icon={['fab', 'github']} />
        </a>
        <p className={styles.name}>
          <FormattedMessage id={nameId} />
        </p>
        <p className={styles.description}>
          <FormattedMessage id={descriptionId} />
        </p>
      </div>
    </div>
  );
}

Card.propTypes = {
  nameId: PropTypes.string.isRequired,
  descriptionId: PropTypes.string.isRequired,
  githubLink: PropTypes.string.isRequired,
};
