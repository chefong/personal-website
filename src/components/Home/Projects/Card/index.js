import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { FormattedMessage } from 'text-provider';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

export default function Card({ nameId, descriptionId, githubLink }) {
    return (
        <li className={styles.cardInfo}>
            <a href={githubLink} target="_blank" rel="noreferrer">
                <FontAwesomeIcon
                    className={styles.github}
                    icon={['fab', 'github']}
                />
            </a>
            <p className={styles.name}>
                <FormattedMessage id={nameId} />
            </p>
            <p className={styles.description}>
                <FormattedMessage id={descriptionId} />
            </p>
        </li>
    );
}

Card.propTypes = {
    nameId: PropTypes.string.isRequired,
    descriptionId: PropTypes.string.isRequired,
    githubLink: PropTypes.string.isRequired,
};
