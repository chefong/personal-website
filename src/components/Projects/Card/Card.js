import styles from './Card.module.css';

export default function Card({ name, description, githubLink }) {
  return (
    <div className="col-md-4">
      <div className={styles.cardInfo}>
        <a href={githubLink} target="_blank">
          <img className={styles.github} src="/imgs/projects/github.svg" alt=""/>
        </a>
        <p className={styles.name}>{name}</p>
        <p className={styles.description}>{description}</p>
      </div>
    </div>
  )
}