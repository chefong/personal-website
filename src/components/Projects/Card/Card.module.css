.cardInfo {
  padding: 32px;
  padding-top: 36px;
  color: #303030;
  border: 2px solid #303030;
  border-radius: 8px;
  height: 208px;
  margin-bottom: 2em;
  transition: all 0.25s ease;
  box-shadow: -12px 12px 0px #303030;
}

.cardInfo:hover {
  box-shadow: none;
  transform: translate(-12px, 12px)
}

.name {
  font-weight: 500;
  font-size: 1.2em;
  margin-bottom: 8px;
}

.github {
  width: 18px;
  float: right;
}