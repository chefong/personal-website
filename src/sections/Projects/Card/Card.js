import styles from './Card.module.css';
import { Badge } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export default function Card({ name, description, githubLink, wonHackathon }) {
  return (
    <div className="col-lg-4 col-md-6">
      <div className={styles.cardInfo}>
        <a href={githubLink} target="_blank">
          <FontAwesomeIcon className={styles.github} icon={['fab', 'github']} />
        </a>
        <p className={styles.name}>{name}</p>
        <p className={styles.description}>{description}</p>
        {wonHackathon && <Badge className={styles.badge}>HACKATHON WINNER</Badge>}
      </div>
    </div>
  )
}