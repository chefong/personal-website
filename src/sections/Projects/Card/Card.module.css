.cardInfo {
  padding: 32px;
  padding-top: 36px;
  color: #303030;
  border: 2px solid #303030;
  border-radius: 8px;
  height: 248px;
  margin-bottom: 2em;
  transition: all 0.25s ease;
  box-shadow: -12px 12px 0px #303030;
  position: relative;
}

.cardInfo:hover {
  box-shadow: none;
  transform: translate(-12px, 12px)
}

.name {
  font-weight: 500;
  font-size: 1.2em;
  margin-bottom: 8px;
}

.github {
  float: right;
  width: 1.25em;
  color: #303030;
  transition: all 0.25s ease;
}

.github:hover {
  color: #4A74E1;
}

.badge {
  background-color: #4A74E1;
  position: absolute;
  bottom: 32px;
}