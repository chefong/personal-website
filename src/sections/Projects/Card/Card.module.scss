.cardInfo {
  padding: 32px;
  padding-top: 36px;
  color: var(--theme-page-text);
  border: 2px solid var(--theme-page-text);
  border-radius: 8px;
  height: 248px;
  margin-bottom: 2em;
  transition: all 0.25s ease;
  box-shadow: -12px 12px var(--theme-page-text);
  position: relative;
}

.cardInfo:hover {
  box-shadow: none;
}

.name {
  font-weight: 500;
  font-size: 1.2em;
  margin-bottom: 8px;
}

.github {
  float: right;
  width: 1.25em;
  color: var(--theme-page-text);
  transition: all 0.25s ease;
}

.github:hover {
  color: var(--theme-color);
}

.badge {
  background-color: var(--theme-color);
  position: absolute;
  bottom: 32px;
}