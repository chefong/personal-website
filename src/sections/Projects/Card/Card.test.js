import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

describe('Card', () => {
  it('renders', () => {
    const { getByTestId } = render(<Card />);
    expect(getByTestId('Card-container')).toBeDefined();
  });

  it('matches snapshot', () => {
    const { container } = render(<Card />);
    expect(container).toMatchSnapshot();
  });
});