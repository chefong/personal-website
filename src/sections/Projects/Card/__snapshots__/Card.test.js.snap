// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
<div>
  <div
    class="col-lg-4 col-md-6"
  >
    <div
      data-testid="Card-container"
    >
      <a
        target="_blank"
      />
      <p>
        <span />
      </p>
      <p>
        <span />
      </p>
    </div>
  </div>
</div>
`;
